### PR TITLE
fix: Removed TSLint comments & refs in ZE package

### PR DESCRIPTION
**Pull request by traeok**
## Proposed changes

- Removes TSLint itself, and any TSLint comments and references within `zowe-explorer` package. This is because ESLint is now recommended and TSLint is deprecated.

## Release Notes

Milestone: 2.5.0

Changelog: Removed TSLint for `zowe-explorer` package

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)

## Checklist

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [ ] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found

## Further comments

- A few questions about rule translation:
  - `one-line` doesn't seem to have a direct replacement - will `curly` suffice?
  - `no-implicit-dependencies` is supposedly covered in `eslint-plugin-node`, although I'm not sure what rules they are added under. Any concerns w/ adding this plugin?

~Here's a translation map that I created for reference:~
*Update:* Turns out most of those options were being explicitly disabled within our TSLint config, so I've removed most of the table. Since the table really isn't applicable anymore, feel free to ignore it.

| TSLint option | ESLint option/alternatives |
| --------------- | ---------------------------- |
| no-unused-expression | @typescript-eslint/no-unused-expressions OR ESLint's no-unused-expressions? |
| &emsp;- allow-fast-null-checks | not in ESLint, but in TS-ESLint lib |
| max-line-length | max-len |
| no-magic-numbers | no-magic-numbers |
| no-consecutive-blank-lines | no-multiple-empty-lines |

### CODE REVIEW

**Comment by traeok** [comment]
Going to shortly reassign as a draft - I want to make sure I didn't miss any rules from the `tslint:latest` ruleset.

File: .eslintrc.yaml
Comment (zFernand0) [inline .eslintrc.yaml:18]: Q: 150 vs 120?
I'll be more than happy to increase the prettier printWidth to 150 🙏 
https://github.com/zowe/vscode-extension-for-zowe/blob/383e25cdfd6f5ee2bfb9b9e0a8c666fb47e1733f/prettier.config.js#L1

**Review by zFernand0 — COMMENTED** [review]
Should the lint workflow flag the following problems?
```
[_eslint_] /home/runner/work/vscode-extension-for-zowe/vscode-extension-for-zowe/packages/zowe-explorer-ftp-extension/src/ZoweExplorerFtpJesApi.ts
[_eslint_]   98:30  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
[_eslint_] 
[_eslint_] /home/runner/work/vscode-extension-for-zowe/vscode-extension-for-zowe/packages/zowe-explorer-ftp-extension/src/ZoweExplorerFtpMvsApi.ts
[_eslint_]   409:33  warning  Unsafe argument of type `any` assigned to a parameter of type `BinaryLike`  @typescript-eslint/no-unsafe-argument
[_eslint_] 
[_eslint_] /home/runner/work/vscode-extension-for-zowe/vscode-extension-for-zowe/packages/zowe-explorer-ftp-extension/src/ZoweExplorerFtpUssApi.ts
[_eslint_]   [27](https://github.com/zowe/vscode-extension-for-zowe/actions/runs/3708091102/jobs/6285271165#step:5:28)1:[33](https://github.com/zowe/vscode-extension-for-zowe/actions/runs/3708091102/jobs/6285271165#step:5:34)  warning  Unsafe argument of type `any` assigned to a parameter of type `BinaryLike`  @typescript-eslint/no-unsafe-argument
```

Also, can we fix this final warnings

> /home/runner/work/vscode-extension-for-zowe/vscode-extension-for-zowe/packages/zowe-explorer/src/dataset/actions.ts
Warning:   306:1  warning  This line has a length of 155. Maximum allowed is 150  max-len
> /home/runner/work/vscode-extension-for-zowe/vscode-extension-for-zowe/packages/zowe-explorer/src/uss/ZoweUSSNode.ts
Warning:   [48](https://github.com/zowe/vscode-extension-for-zowe/actions/runs/3708091102/jobs/6285271165#step:5:49)6:1  warning  This line has a length of 162. Maximum allowed is 150  max-len
> /home/runner/work/vscode-extension-for-zowe/vscode-extension-for-zowe/packages/zowe-explorer/src/uss/actions.ts
Warning:   85:1  warning  This line has a length of 159. Maximum allowed is 1[50](https://github.com/zowe/vscode-extension-for-zowe/actions/runs/3708091102/jobs/6285271165#step:5:51)  max-len
> /home/runner/work/vscode-extension-for-zowe/vscode-extension-for-zowe/packages/zowe-explorer/src/utils/SettingsConfig.ts
Warning:   73:1  warning  This line has a length of 1[53](https://github.com/zowe/vscode-extension-for-zowe/actions/runs/3708091102/jobs/6285271165#step:5:54). Maximum allowed is 150  max-len

**Comment by traeok** [comment]
> SonarCloud Quality Gate failed.

These security hotspots and code smells already existed, but they were detected again after the formatting changes. I will review these security hotspots as necessary, but the code smells and duplication can be ignored (and as there are a few different code smells/duplication instances, this could be extended to a different PR)

**Review by zFernand0 — CHANGES_REQUESTED** [review]
Before we get this PR merged, I would like to make sure that the Squad is OK with going to a `printWidth: 150`

I'm personally in favor of such change.

File: packages/zowe-explorer/package.json
Comment (t1m0thyj) [inline packages/zowe-explorer/package.json:1771]: Should we keep the separate "lint:tests" script in case anyone was using it?

File: packages/zowe-explorer/src/dataset/actions.ts
Comment (t1m0thyj) [inline packages/zowe-explorer/src/dataset/actions.ts:266]: Should this string be split across multiple lines to make the linter happy?

File: packages/zowe-explorer/src/uss/USSTree.ts
Comment (t1m0thyj) [inline packages/zowe-explorer/src/uss/USSTree.ts:763]: Should we rename `intializeUSSFavorites` -> `initializeUSSFavorites` to fix the typo? We could create a separate issue for this as well, since I know it's outside the scope of this PR 🙂

File: packages/zowe-explorer/src/uss/ZoweUSSNode.ts
Comment (t1m0thyj) [inline packages/zowe-explorer/src/uss/ZoweUSSNode.ts:457]: Same question as above - should we split across multiple lines to make the linter happy?

File: packages/zowe-explorer/src/uss/actions.ts
Comment (t1m0thyj) [inline packages/zowe-explorer/src/uss/actions.ts:80]: Same question as above - should we split across multiple lines to make the linter happy?

File: packages/zowe-explorer/src/utils/ProfilesUtils.ts
Comment (t1m0thyj) [inline packages/zowe-explorer/src/utils/ProfilesUtils.ts:30]: We could also import `RestConstants.HTTP_STATUS_401` from Imperative to use that constant 🤷  https://github.com/zowe/imperative/blob/ca981d476b7a97fa6129de523c63e1a58000c139/packages/rest/src/client/RestConstants.ts#L66

File: packages/zowe-explorer/src/utils/SettingsConfig.ts
Comment (t1m0thyj) [inline packages/zowe-explorer/src/utils/SettingsConfig.ts:61]: Same question as above - should we split line 67 across multiple lines to make the linter happy?

File: packages/zowe-explorer/src/uss/USSTree.ts
Comment (traeok) [reply packages/zowe-explorer/src/uss/USSTree.ts:763]: Fixed in new commit 👍

File: packages/zowe-explorer/src/dataset/actions.ts
Comment (traeok) [reply packages/zowe-explorer/src/dataset/actions.ts:266]: Just discussed w/ @t1m0thyj that the localize functions do not like any parameters other than raw string literals; unfortunately backticks introduce unwanted spaces in the localized strings, so that approach will not work either. Disabled lint warnings for the lines mentioned.

File: packages/zowe-explorer/src/uss/actions.ts
Comment (traeok) [reply packages/zowe-explorer/src/uss/actions.ts:80]: See above comment regarding lint warnings

File: packages/zowe-explorer/src/utils/SettingsConfig.ts
Comment (traeok) [reply packages/zowe-explorer/src/utils/SettingsConfig.ts:61]: See above comment regarding lint warnings

File: packages/zowe-explorer/src/uss/ZoweUSSNode.ts
Comment (traeok) [reply packages/zowe-explorer/src/uss/ZoweUSSNode.ts:457]: See above comment regarding lint warnings

File: packages/zowe-explorer/package.json
Comment (traeok) [reply packages/zowe-explorer/package.json:1771]: Fixed in new commit 👍

File: .eslintrc.yaml
Comment (JillieBeanSim) [reply .eslintrc.yaml:32]: I do like the addition of the "no-console" error added.  Maybe a few others as well like the "no-duplicate-imports"

**Comment by JillieBeanSim** [comment]
@traeok could we try to resolve the code smells in this PR please

**Comment by traeok** [comment]
> @traeok could we try to resolve the code smells in this PR please

The code smells brought up in this PR were due to reformatting with ESLint and removing the "disable TSLint" lines throughout the codebase. I can work on refactoring these if needed, but as these code smells in this PR are unrelated to the ESLint changes, it would probably be best to separate these into a new issue.

**Review by JillieBeanSim — APPROVED** [review]
@traeok thanks for the update.
I noticed the new .eslintrc.yaml is top level vs within the zowe-explorer package.  The other packages have their own .eslintrc.yaml file.  I can add this info into the [issue](https://github.com/zowe/vscode-extension-for-zowe/issues/2068) I created for the discussion of the eslint rules we would like to throw errors and we can have the rules at a global level and remove the others from within the projects.

**Comment by traeok** [comment]
> @traeok thanks for the update. I noticed the new .eslintrc.yaml is top level vs within the zowe-explorer package. The other packages have their own .eslintrc.yaml file. I can add this info into the [issue](https://github.com/zowe/vscode-extension-for-zowe/issues/2068) I created for the discussion of the eslint rules we would like to throw errors and we can have the rules at a global level and remove the others from within the projects.

@JillieBeanSim Thanks for making that issue - I created an issue yesterday regarding the code duplication/code smell warnings that came up from these PRs, and can add it to the PI planning if desired.

And my apologies for not addressing those code smells w/ the `jest.spyOn` calls (and thanks for fixing those!) - I tried fixing them locally yesterday but I must have overlooked something as I broke a couple tests on my end.

**Comment by JillieBeanSim** [comment]
> > @traeok thanks for the update. I noticed the new .eslintrc.yaml is top level vs within the zowe-explorer package. The other packages have their own .eslintrc.yaml file. I can add this info into the [issue](https://github.com/zowe/vscode-extension-for-zowe/issues/2068) I created for the discussion of the eslint rules we would like to throw errors and we can have the rules at a global level and remove the others from within the projects.
> 
> @JillieBeanSim Thanks for making that issue - I created an issue yesterday regarding the code duplication/code smell warnings that came up from these PRs, and can add it to the PI planning if desired.
> 
> And my apologies for not addressing those code smells w/ the `jest.spyOn` calls (and thanks for fixing those!) - I tried fixing them locally yesterday but I must have overlooked something as I broke a couple tests on my end.

no worries and thanks for creating those issues.  please go ahead and add them to the Plan for us

**Review by zFernand0 — APPROVED** [review]

**Comment by JillieBeanSim** [comment]
@traeok could you address the merge conflicts in this PR please.  Looks like we got the approvals needed.

**Review by zFernand0 — APPROVED** [review]

**Review by t1m0thyj — APPROVED** [review]
LGTM! I suggest we merge this ASAP since this PR is probably one of the more painful ones to keep up to date and re-review 😀

**Comment by zFernand0** [comment]
I was thinking the same thing. 
- 😋 This can wait: #2038

**Pull request merged**
